### ISSUE TITLE
Moving a mail to a previously deleted folder vanishes moved mail

### ISSUE THREAD

**Issue by BeatBurgener** [RealRaven2000/QuickFolders#515]
Deleted a folder, which will still show up in the QF QuickMove/QuickJump list. Unfortunately moved the mail to the now non existing folder ... mail appears to be gone ... undo does  not work with QF ... or I don't know how ... QF 6.8.2 / TB 128.5.2esr (64bit)

**Comment by RealRaven2000** [comment]
> Deleted a folder, which will still show up in the QF QuickMove/QuickJump list. Unfortunately moved the mail to the now non existing folder ... mail appears to be gone ... undo does not work with QF ... or I don't know how ... QF 6.8.2 / TB 128.5.2esr (64bit)

ok, that's a bit of a  problem - if you delete a folder, it may end up in the "Trash" folder, and the link may be redirected there by Thunderbird, but let me confirm this - I will look into it. If you do not auto-empty your trash on Thunderbird exit it will still be sitting there in the (now deleted) folder.

I think the least I can do is generate a prompt (when moving) that checks on whether one of the parent folders of the target has the "Trash" flag set. Note that people my actually want that behavior though... so we would need to make this prompt configurable. Another thing would be to show a "trash" icon in front of such folders in the list.

**Comment by BeatBurgener** [comment]
Axel, thank you for the swift reply and sorry, this was a "false alert" ... the "problem" is TB, not QF. Search did not return the "lost" mail - maybe this is the default behaviour of TB to not show results from the special folder "Deleted". Plus I did not pay attention in which mailbox the deleted folder was, and did check the wrong mailbox for that folder.
At the end of the day, the moved mail was still available in the "Deleted/Subfolder-XYZ" folder. All good.
I further tested to delete this subfolder-XYZ from the "Deleted" folder, and QF immediately stopped showing this as a jump/move target. 
Again, sorry for taking your time for this none-issue. Keep up the great work - cannot live without that Addon!

**Comment by RealRaven2000** [comment]
> At the end of the day, the moved mail was still available in the "Deleted/Subfolder-XYZ" folder. All good.

yeah, still wondering whether we should (during a quickJump search) either:
- hide all folders / subfolders contained within a Trash folder
or:
- add it to the list but add some "Trash" icon and / or highlight the folder name in some other way (e.g. crossed out / red text)

**Comment by BeatBurgener** [comment]
Looking at it from my perspective, if a folder was deleted, there is likely nothing you would like to move to that folder anyway, or there is nothing left to move there as the contents of the deleted folder was deleted with it. E.g. I don't see the use case for the option to move a mail to a sub-folder of trash, which in turn favors your first proposal to not show folders from the trash.

**Comment by RealRaven2000** [comment]
> Looking at it from my perspective, if a folder was deleted, there is likely nothing you would like to move to that folder anyway, or there is nothing left to move there as the contents of the deleted folder was deleted with it. E.g. I don't see the use case for the option to move a mail to a sub-folder of trash, which in turn favors your first proposal to not show folders from the trash.

That is correct, it would almost always be one of 2 other cases:

- the target folder was accidentally deleted and the user is not aware of that. In this case a warning would be preferable.
- the target folder was mistyped and you really wanted to move the mail to a completely different folder.

Pros and cons of completely hiding the folder:
- if the folder was deleted accidentally, a warning would be nice!

**Comment by BeatBurgener** [comment]
... I would then tend more for the visual "hint", e.g. folder being in Grey color (might be difficult based on theme) or "strike through", which might work with any color ...

**Comment by RealRaven2000** [comment]
> ... I would then tend more for the visual "hint", e.g. folder being in Grey color (might be difficult based on theme) or "strike through", which might work with any color ...

I could gray it out (or even disable it) - but it might just frustrate users unless I can provide an explanation when they click on the entry:

The folder _Name_ was deleted, do you want to navigate there to check? [OK] [Cancel]

**Comment by BeatBurgener** [comment]
True, as mentioned: in which use case would a user want to jump to /move to a folder in trash? Jump to maybe, move to less likely ... If technically possible, displaying the trash icon in front of folder that is currently in trash might be an option or grayed out alternatively. 

For the jump action, no message is required IMHO, for the move action - if possible - a warning might be useful. From an usability perspective, a "never show this message again" option would then be required IMHO.

**Comment by RealRaven2000** [comment]
> For the jump action, no message is required IMHO, for the move action - if possible - a warning might be useful. From an usability perspective, a "never show this message again" option would then be required IMHO.


Like this?

Move the mail anyway? [Yes] [No]
[x] Do not show this message again. 

activating the option with [Yes] would mean it will always move mail to deleted folders without asking again. Might not be a good idea, some people just click and check stuff without thinking... I will consider it.
